overlord/ifacestate/udevmonitor/udevmon.go

```diff
@@ -104,46 +106,31 @@ func (m *Monitor) Disconnect() error {
 // handles hotplug events (devices added or removed). It returns immediately.
 // The goroutine must be stopped by calling Stop() method.
 func (m *Monitor) Run() error {
+	// Gather devices from udevadm info output (enumeration on startup).
+	devices, parseErrors, err := hotplug.EnumerateExistingDevices()
+	if err != nil {
+		return fmt.Errorf("cannot enumerate existing devices: %s", err)
+	}
 	m.tomb.Go(func() error {
-		existingDevices := make(chan *hotplug.HotplugDeviceInfo)
-		udevadmErrors := make(chan error)
-
-		// Gather devices from udevadm info output (enumeration on startup).
-		if err := hotplug.EnumerateExistingDevices(existingDevices, udevadmErrors); err != nil {
-			return fmt.Errorf("cannot enumerate existing devices: %s", err)
+		for _, perr := range parseErrors {
+			logger.Noticef("udev enumeration error: %q\n", perr)
```

Comment:
This needs to be %s or %v, not %q.

Reply:
Thanks, fixed (also another instance of this below); also removed newlines from these errors, not sure why did I have them.